partitioner-config-service-impl/src/main/java/org/hypertrace/partitioner/config/service/PartitionerConfigServiceRequestValidator.java

```diff
@@ -0,0 +1,75 @@
+package org.hypertrace.partitioner.config.service;
+
+import io.grpc.Status;
+import java.util.HashMap;
+import java.util.Map;
+import org.hypertrace.partitioner.config.service.v1.*;
+
+public class PartitionerConfigServiceRequestValidator {
+
+  public void validateOrThrow(GetPartitionerProfileRequest request) {
+    if (request.getProfile().isBlank()) {
+      throw Status.INVALID_ARGUMENT.withDescription("profile cannot be empty").asRuntimeException();
+    }
+  }
+
+  public void validateOrThrow(PutPartitionerConfigRequest request) {
+    PartitionerProfile profile = request.getProfile();
+    if (profile.getName().isBlank()) {
+      throw Status.INVALID_ARGUMENT.withDescription("profile cannot be empty").asRuntimeException();
+    }
+    if (profile.getGroupsCount() == 0) {
+      throw Status.INVALID_ARGUMENT
+          .withDescription("partition_groups cannot be empty")
+          .asRuntimeException();
+    }
+    if (profile.getPartitionKey().isBlank()) {
+      throw Status.INVALID_ARGUMENT
+          .withDescription("partition key cannot be empty")
+          .asRuntimeException();
+    }
+    profile
+        .getGroupsList()
+        .forEach(
+            partitionerGroup -> {
+              if (partitionerGroup.getName().isBlank()) {
+                throw Status.INVALID_ARGUMENT
```

Comment:
this one was informational only right? So it _could_ be. If you don't want it to though, that's fine.

Reply:
keeping it for now